docs/Reboots.md

```diff
@@ -0,0 +1,29 @@
+# What causes reboots?
+
+Under certain conditions, MachineConfig updates will cause the Machine
+Config Daemon to reboot the nodes affected by the change. This document
+aims to identify exactly what MachineConfig changes cause a reboot, with
+the intent that any changes will not be unexpected.
+
+A MachineConfig change will cause a reboot under the following
+conditions:
```

Comment:
This sentence isn't quite correct as an os update isn't really a machine config change initiated by the user in the same ways. Maybe make this more generic "The following will cause a a node to reboot"